src/server/utils/tracer.js

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// The NoOpTracer is a close variant to the Tracer and so is ok in this file
+// eslint-disable-next-line max-classes-per-file
+export class Tracer {
```

Comment:
Object Orientation here offers an explicit advantage, Encapsulation and Abstraction. 

By encapsulating all the tracing functions into the Tracer class, we can (very syntactically cleanly) also create a NoOpTracer.

This allows the server to 'install' either Tracer or NoOpTracer, and all the other code in the system does not know the difference.

That way the check to ONE_ENABLE_SERVER_TRACER can be singular and central.

This also allows for the clean use of private fields and functions, and there is no ambiguity over the scope of all memory allocated by this system, as it can be garbage collected all at once.

Reply:
encapsulation and easy to access to fns make sense in this context, I'm just always get a bit worried about contributors getting too passionate about classes and start extending them, making solutions unreadable and hard to maintain